converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/ImagingStudy/ImagingStudyInstancePropertySynchronizer.cs

```diff
@@ -26,14 +44,54 @@ public void Synchronize(FhirTransactionContext context, ImagingStudy.InstanceCom
                 return;
             }
 
-            // Add sopclass to instance
+            SynchronizePropertiesAsync(instance, context, true, AddSopClass, cancellationToken);
+            SynchronizePropertiesAsync(instance, context, false, AddInstanceNumber, cancellationToken);
+        }
+
+        private void SynchronizePropertiesAsync(ImagingStudy.InstanceComponent instance, FhirTransactionContext context, bool required, Action<ImagingStudy.InstanceComponent, FhirTransactionContext> synchronizeAction, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                synchronizeAction(instance, context);
+            }
+            catch (Exception ex)
+            {
+                if (_dicomValidationConfiguration.PartialValidation && !required)
+                {
+                    DicomDataset dataset = context.ChangeFeedEntry.Metadata;
+                    string studyUID = dataset.GetSingleValue<string>(DicomTag.StudyInstanceUID);
+                    string seriesUID = dataset.GetSingleValue<string>(DicomTag.SeriesInstanceUID);
+                    string instanceUID = dataset.GetSingleValue<string>(DicomTag.SOPInstanceUID);
+
+                    _exceptionStore.StoreException(
+                        studyUID,
+                        seriesUID,
+                        instanceUID,
+                        context.ChangeFeedEntry.Sequence,
+                        ex,
+                        ErrorType.DicomValidationError,
```

Comment:
How are we categorizing this has validationError? We are catching all exceptions here

Reply:
So my thought here is that each time we are executing the Action we are trying to sync a specific property, and if while doing that on a field that is not required we can just store the error and continue to the next one. 

Technically these all aren't validation errors but if it is not a required field can't we continue processing regardless? Or do we want to handle those errors differently?

Reply:
Updated to catch more specific validation error.